proto/raft_serverpb.proto

```diff
@@ -60,6 +60,59 @@ message SnapshotChunk {
 
 message Done {}
 
+message TabletSnapshotFileMeta {
+    uint64 file_size = 1;
+    string file_name = 2;
+    // Some block data.
+    bytes head_chunk = 3;
+    // trailing data including checksum.
+    bytes trailing_chunk = 4;
+}
+
+// Snapshot preview for server to decide whether skip some files.
+// Server should send back an `AcceptedSnapshotFile` to let client
+// keep sending specified files. Only SST files can be skipped, all
+// other files should always be sent.
+message TabletSnapshotPreview {
+    repeated TabletSnapshotFileMeta metas = 1;
+    // There may be too many metas, use a flag to indicate all metas
+    // are sent.
+    bool end = 2;
+}
+
+message TabletSnapshotFileChunk {
+    uint64 file_size = 1;
+    string file_name = 2;
+    bytes data = 3;
+}
+
+message TabletSnapshotHead {
+    RaftMessage message = 1;
+    bool use_cache = 2;
+}
+
+message TabletSnapshotEnd {
+    // Checksum of all data sent in `TabletSnapshotFileChunk.data`.
+    uint64 checksum = 1;
```

Comment:
There still be chance in software bugs or hardware errors. We have observed message corruption in the application layer (like raft message) in practice. Note the overhead is small. According to https://github.com/tikv/crc64fast, it can process more than 28GiB per second. The major overhead is from IO.